types/test/toArray.ts

```diff
@@ -1,7 +1,10 @@
 import * as RA from 'ramda-adjunct';
 
-RA.toArray([1, 2, 3]); // $ExpectType any[]
-RA.toArray(null); // $ExpectType any[]
-RA.toArray(undefined); // $ExpectType any[]
-RA.toArray(1); // $ExpectType any[]
-RA.toArray({}); // $ExpectType any[]
+RA.toArray('1'); // $ExpectType string[]
+RA.toArray([1, 2, 3]); // $ExpectType number[]
+RA.toArray({ a: 1, b: 2 }); // $ExpectType number[]
+RA.toArray({}); // $ExpectType never[]
```

Comment:
In my understanding `never` is a type that we use for functions that never returns, for example functions that always throws. I also understand that `never` is subtype of every other type and is assignable to every other type. But still why we use `never[]` here eludes me ;]

Reply:
Sorry for the delayed response. Btw, good observation! The reason for this is that since there are no keys/values in object, TS can't infer which type the `array` is and therefore assumes `never`.

Note that it's the same behavior of [**`ramda#keys`**](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts#L895-L896), which also returns `never` for empty objects.

<hr>

When I looked at this test the first time I was wondering what was the real benefit of having this... I mean what do we expect when passing an empty object? An empty array? So why not just write `[]` instead of `toArray({})`? Anyway, at that time I preferred to keep this and just change the `ExpectType` to conform the stronger typing, but **now** I see it could be easily removed, don't you agree?

<hr>

The playgrounds for testing:

[**With empty object**](https://www.typescriptlang.org/play/#code/C4TwDgpgBAaghgGwK4QPIDMA8AVAfFAXimwG0BrCEAe3WIF0BuAKCYBMIBjBOAJ2nSQA7DsACWVQVGBUAgjx5wQOKBAAewCINYBnKFQBGAK07BcAClEaF+hBABcxAJQP4yNFjwlGLFtLkKQMwBvAF9HIA)
[**With filled object**](https://www.typescriptlang.org/play/#code/C4TwDgpgBAaghgGwK4QPIDMA8AVAfFAXimwG0BrCEAe3WIF0BuAKCYBMIBjBOAJ2nSQA7DsACWVQVGBUAgjx5wQOKBAAewCINYBnKFQBGAK07BcAClEaF+hBABcxAJQP4yNFjwlGLHy2lyFEDMAbzgHAEYAGih9BwByACY4gF9HIA)